Library/Homebrew/exceptions.rb

```diff
@@ -87,21 +87,31 @@ def initialize(name)
     super()
 
     @name = name
+    @without_api = Homebrew::EnvConfig.no_install_from_api?
+    @auto_without_api = Homebrew::EnvConfig.automatically_set_no_install_from_api?
```

Comment:
Will this still be correct when queried inside `to_s` where `without_api` may have changed the value?

Reply:
We care about the state of when the exception was thrown, which might have been inside a `Homebrew.without_api` block, rather then whenever we get around to printing it.

Reply:
Gotcha. Does that apply for both of them? Feels potentially confusing given `Homebrew.without_api` is used below but could just be resolved with a (Ruby) comment rather than anything more involved. Thanks!

Reply:
Yeah. The use below is so if we couldn't find a tap in api or no-api mode, then we also perform a fuzzy search for the "did you mean" under that same mode.

Reply:
Gotcha. Yeh, comment would clear that up!